include/rtos/task.h

```diff
@@ -1749,6 +1749,24 @@ int32_t task_notify_wait( uint32_t ulBitsToClearOnEntry, uint32_t ulBitsToClearO
  */
 int32_t task_notify(task_t xTaskToNotify);
 
+/**
+ * task. h
+ * <PRE>void task_join( task_t taskToJoin );</PRE>
+ *
+ * configUSE_TASK_NOTIFICATIONS must be undefined or defined as 1 for this macro
+ * to be available.
+ * 
+ * Waits until specified task is complete and deleted, then continues to execute the program.
+ * Analogous to std::thread::join in C++.
+ *
+ * @param xTaskToJoin The handle of the task to wait on.
+ * 
+ * \defgroup task_join task_join
+ * \ingroup TaskNotifications
+ */
```

Comment:
no need to copy freeRTOS style documentation here, the important part is in pros/rtos.h

and actually this declaration isn't necessary at all...